models/dns_provider.go

```diff
@@ -1,36 +1,85 @@
 package models
 
 import (
+	"code.cloudfoundry.org/lager"
+	"context"
+	"crypto/rsa"
+	"fmt"
+	"gopkg.in/square/go-jose.v1"
 	"time"
 
 	"github.com/18F/cf-cdn-service-broker/config"
-	"github.com/18F/cf-cdn-service-broker/lego/acme"
 	"github.com/18F/cf-cdn-service-broker/utils"
+
+	legoacme "github.com/18F/cf-cdn-service-broker/lego/acme"
+	goacme "golang.org/x/crypto/acme"
 )
 
-func (*AcmeClientProvider) GetDNS01Client(user *utils.User, settings config.Settings) (acme.ClientInterface, error) {
-	client, err := acme.NewClient(settings.AcmeUrl, user, acme.RSA2048)
+func (acp *AcmeClientProvider) GetDNS01Client(user *utils.User, settings config.Settings) (legoacme.ClientInterface, error) {
+	logSess := acp.logger.Session("get-dns-01-client")
 
-	if err != nil {
-		return &acme.Client{}, err
+	logSess.Info("create-goacme-client")
+	key := user.GetPrivateKey().(*rsa.PrivateKey)
+	client := goacme.Client{Key: key}
+
+	ctx := context.Background()
+
+	logSess.Info("create-goacme-account-struct")
+	a := goacme.Account{
+		Contact: []string {fmt.Sprintf("mailto:%s", user.Email)},
 	}
 
-	if user.GetRegistration() == nil {
-		reg, err := client.Register()
+	logSess.Info("fetch-goacme-account")
+	account, err := client.GetReg(ctx, "this argument is ignored because the CA is RFC8555 compliant. The key on the client is used instead.")
+	if err == goacme.ErrNoAccount {
+		logSess.Info("goacme-account-not-found")
+		logSess.Info("register-goacme-account")
+		account, err = client.Register(ctx, &a, goacme.AcceptTOS)
 		if err != nil {
-			return client, err
+			logSess.Error("register-goacme-account-error", err)
+			return nil, err
+		}
+	} else if err != nil {
+		logSess.Error("fetch-goacme-account", err)
```

Comment:
Should we be returning the error here? Seems like it's unrecoverable.

Reply:
Yes we probably should be. That's an oversight on my part.

Reply:
Have resolved this and force pushed